sympy/polys/polyclasses.py

```diff
@@ -2136,7 +2143,7 @@ def _cancel(f, g):
         elif f_neg:
             cF, F = -cF, F.neg()
         elif g_neg:
-            cG, G = -cG, G.neg()
+            cF, G = -cF, G.neg()
```

Comment:
A bug in the `cancel` method. See the equivalent code:
https://github.com/sympy/sympy/blob/c6c70fa525de648d58872261de02d92fac81c7f9/sympy/polys/euclidtools.py#L1896-L1904